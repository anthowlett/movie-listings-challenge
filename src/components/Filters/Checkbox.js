import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { colors } from '../../utils';

const Wrapper = styled.span`
  input {
    opacity: 0;
    position: absolute; 
  }

  label {
    display: inline-block;
    position: relative;
    margin: 0 5px 10px;
    padding: 7px;
    vertical-align: middle;
    font-size: 14px;
    cursor: pointer;
    border: 1px solid black;
    transition: background-color 0.25s;
    border-radius: 5px;
  }

  input:checked + label {
    color: white;
    background-color: ${colors.green};
    border-color: green;
  }
`;

const Checkbox = ({
	id,
	name,
	checked,
	onChange
}) => (
  <Wrapper>
    <input
      id={id}
      type='checkbox'
      name='list'
      value={id}
      checked={checked}
      onChange={(e) => onChange(parseInt(e.target.value), e.target.checked)}
    />
    <label htmlFor={id}>{name}</label>
  </Wrapper>
);

Checkbox.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Checkbox;