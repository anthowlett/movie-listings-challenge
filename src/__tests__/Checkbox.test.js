import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '../components/Filters/Checkbox';
import '../setupTests';

describe('Filters', () => {

  describe('<Checkbox />', () => {
    const id = 1;
    const name = 'Genre name';
    const onChange = jest.fn();
    const checked = false;
    const component = <Checkbox key={id} id={id} name={name} checked={checked} onChange={onChange} />
    const tree = renderer.create(component).toJSON();

    it('should render', () => {
      expect(tree)
        .toMatchSnapshot();
    });

    it('should contain an input', () => {
      expect(tree.children.map(child => child.type))
        .toContain('input');
    });

    it('should contain a label', () => {
      expect(tree.children.map(child => child.type))
        .toContain('label');
    });

    it('label should contain text', () => {
      const label = tree.children.filter(child => child.type === 'label')[0];
      expect(label.children[0]).toEqual(
        expect.stringMatching(name)
      );
    });
  });

});