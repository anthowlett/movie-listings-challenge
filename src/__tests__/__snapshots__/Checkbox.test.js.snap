// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters <Checkbox /> should render 1`] = `
.c0 input {
  opacity: 0;
  position: absolute;
}

.c0 label {
  display: inline-block;
  position: relative;
  margin: 0 5px 10px;
  padding: 7px;
  vertical-align: middle;
  font-size: 14px;
  cursor: pointer;
  border: 1px solid black;
  -webkit-transition: background-color 0.25s;
  transition: background-color 0.25s;
  border-radius: 5px;
}

.c0 input:checked + label {
  color: white;
  background-color: #00A14D;
  border-color: green;
}

<span
  className="c0"
>
  <input
    checked={false}
    id={1}
    name="list"
    onChange={[Function]}
    type="checkbox"
    value={1}
  />
  <label
    htmlFor={1}
  >
    Genre name
  </label>
</span>
`;
